dags/dynamic-challenge-dag.py

```diff
@@ -0,0 +1,141 @@
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+from orca.services.nextflowtower import NextflowTowerHook
+from orca.services.nextflowtower.models import LaunchInfo
+from orca.services.synapse import SynapseHook
+
+UUID = uuid.uuid4()
+REGION_NAME = "us-east-1"
+BUCKET_NAME = "dynamic-challenge-project-tower-scratch"
+FILE_NAME = f"submissions_{UUID}.csv"
+KEY = "10days/dynamic_challenge"
+
+
+dag_params = {
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "aws_conn_id": Param("AWS_TOWER_PROD_S3_CONN", type="string"),
+    "tower_conn_id": Param("DYNAMIC_CHALLENGE_PROJECT_TOWER_CONN", type="string"),
+    "tower_run_name": Param(f"dynamic-challenge-evaluation_{UUID}", type="string"),
+    "tower_compute_env_type": Param("spot", type="string"),
+    "view_id": Param("syn52658661", type="string"),
+    "testing_data": Param("syn53627077", type="string"),
+    "scoring_script": Param("dynamic_challenge_score.py", type="string"),
+    "validation_script": Param("dynamic_challenge_validate.py", type="string"),    
+    "email_with_score": Param("yes", type="string"),
+    "synapse_evaluation_id": Param("9615537", type="string"),
+}
+
+dag_config = {
+    # Run every 2 minutes
+    "schedule_interval": "*/2 * * * *",
+    "start_date": datetime(2024, 1, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 2,
+    },
+    "tags": ["nextflow_tower"],
+    "params": dag_params,
+}
+
+
+@dag(**dag_config)
+def dynamic_challenge_dag():
+    @task.branch()
+    def check_for_new_submissions(**context):
+        hook = SynapseHook(context["params"]["synapse_conn_id"])
+        # Unfortunately, you cannot use XCom in a branching task
+        # So we have to call `get_submissions_with_status` twice in the DAG
+        submissions = hook.ops.get_submissions_with_status(
+            context["params"]["view_id"], "RECEIVED"
+        )
+        if submissions:
+            return "get_new_submissions"
+        return "stop_dag"
+
+    @task
+    def get_new_submissions(**context):
+        hook = SynapseHook(context["params"]["synapse_conn_id"])
+        submissions = hook.ops.get_submissions_with_status(
+            context["params"]["view_id"], "RECEIVED"
+        )
+        return submissions
+
+    @task()
+    def stop_dag():
+        pass
+
+    @task()
+    def update_submission_statuses(submissions: list, **context):
+        hook = SynapseHook(context["params"]["synapse_conn_id"])
+        for submission in submissions:
+            hook.ops.update_submission_status(
+                submission_id=submission,
+                submission_status="EVALUATION_IN_PROGRESS",
+            )
+
+    @task()
+    def stage_submissions_manifest(submissions: list, **context):
+        s3_hook = S3Hook(
+            aws_conn_id=context["params"]["aws_conn_id"], region_name=REGION_NAME
+        )
+        df = pd.DataFrame({"submission_id": submissions})
+        df.to_csv(FILE_NAME, index=False)
```

Comment:
I am realizing that this should probably be a temp file.

Reply:
Alternatively after upload delete the local copy